hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DiskBalancerManager.java

```diff
@@ -95,40 +101,54 @@ public List<HddsProtos.DatanodeDiskBalancerInfoProto> getDiskBalancerReport(
    * If hosts is null, return status of all datanodes in balancing.
    */
   public List<HddsProtos.DatanodeDiskBalancerInfoProto> getDiskBalancerStatus(
-      Optional<List<String>> hosts, int clientVersion) throws IOException {
-    List<HddsProtos.DatanodeDiskBalancerInfoProto> statusList =
-        new ArrayList<>();
+      Optional<List<String>> hosts,
+      Optional<HddsProtos.DiskBalancerRunningStatus> status,
+      int clientVersion) throws IOException {
     List<DatanodeDetails> filterDns = null;
     if (hosts.isPresent() && !hosts.get().isEmpty()) {
       filterDns = NodeUtils.mapHostnamesToDatanodes(nodeManager, hosts.get(),
           useHostnames);
     }
 
-    for (DatanodeDetails datanodeDetails: nodeManager.getNodes(IN_SERVICE,
-        HddsProtos.NodeState.HEALTHY)) {
-      if (shouldReturnDatanode(filterDns, datanodeDetails)) {
-        double volumeDensitySum =
-            getVolumeDataDensitySumForDatanodeDetails(datanodeDetails);
-        statusList.add(HddsProtos.DatanodeDiskBalancerInfoProto.newBuilder()
-            .setCurrentVolumeDensitySum(volumeDensitySum)
-            .setDiskBalancerRunning(isRunning(datanodeDetails))
-            .setDiskBalancerConf(statusMap.getOrDefault(datanodeDetails,
-                    DiskBalancerStatus.DUMMY_STATUS)
-                .getDiskBalancerConfiguration().toProtobufBuilder())
-            .setNode(datanodeDetails.toProto(clientVersion))
-            .build());
-      }
-    }
-    return statusList;
+    // Filter Running Status by default
+    HddsProtos.DiskBalancerRunningStatus filterStatus =
+        status.orElse(HddsProtos.DiskBalancerRunningStatus.RUNNING);
+
+    List<DatanodeDetails> finalFilterDns = filterDns;
+
+    return nodeManager.getAllNodes().stream()
```

Comment:
We are still looping through all nodes here. In case hosts is not empty we should only loop through nodes provided by the hosts input parameter.

Reply:
Updated the for loop logic.